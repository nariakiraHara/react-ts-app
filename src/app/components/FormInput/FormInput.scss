// @import 'src/app/scss/import/assets';

.formInput {
  // スタイルを書いてください
  width: 70%;
  padding: 10px 15px; /*ボックスを大きくする*/
  font-size: 16px;
  border-radius: 6px; /*ボックス角の丸み*/
  border: 2px solid #ddd; /*枠線*/
  box-sizing: border-box; /*横幅の解釈をpadding, borderまでとする*/
}
