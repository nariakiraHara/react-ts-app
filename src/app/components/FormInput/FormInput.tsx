import { FC } from 'react'

import './FormInput.scss'

interface Props {
  /**
   * フォームのどのタイプにするか
   */
  type: 'email' | 'password' | 'text'
  /**
   * placeholder
   */
  placeholder?: string
}

/**
 * FormInputの共通コンポーネント
 */
const FormInput: FC<Props> = ({ type, placeholder }) => {
  return (
    <>
      <input type={type} placeholder={placeholder} className="formInput"/>
    </>
  )
}

export default FormInput
