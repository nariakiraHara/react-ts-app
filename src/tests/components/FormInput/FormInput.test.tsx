import renderer from 'react-test-renderer'
import FormInput from 'app/components/FormInput'

describe('FormInput', () => {
  test('正常系', () => {
    const tree = renderer.create(<FormInput text={'テストテキスト'} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
