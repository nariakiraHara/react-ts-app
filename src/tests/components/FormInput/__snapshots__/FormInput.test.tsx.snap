// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput 正常系 type=email 1`] = `
<input
  className="formInput"
  type="email"
/>
`;

exports[`FormInput 正常系 type=password 1`] = `
<input
  className="formInput"
  type="password"
/>
`;

exports[`FormInput 正常系 type=text 1`] = `
<input
  className="formInput"
  type="text"
/>
`;
